### PR TITLE
Fixes and improvements

### PR DESCRIPTION
**Pull request by faelenor**
See commit comments.

### CODE REVIEW

**Comment by MarkUnity** [comment]
The only thing that needs changes is that the reload is required. If the user alters the asset rules from the proxy assets folder and then the asset auditor window regains focus the changes need to b reflected there.

File: Assets/Editor/AssetAuditor/Scripts/AssetAuditorWindow.cs [hunk 1]
```diff
@@ -284,12 +293,13 @@ void OnSelectionChange()
 
         private void OnFocus()
         {
-            if (m_Initialized)
+            // this doesn't seem to be needed
+            /*if (m_Initialized)
```
Comment (MarkUnity) [inline Assets/Editor/AssetAuditor/Scripts/AssetAuditorWindow.cs:297]: This is needed if changes are made to the assets outside of the tools. When refocusing the window it is good to update the UI.

Comment (faelenor) [reply Assets/Editor/AssetAuditor/Scripts/AssetAuditorWindow.cs:297]: I see... Is there a callback that we could use instead? With projects having 1000s of assets, this was really slow (30s+) for rules matching all assets. We could probably optimize something in the loop as well.

Comment (MarkUnity) [reply Assets/Editor/AssetAuditor/Scripts/AssetAuditorWindow.cs:297]: Ah yeah. Leave it for now I have an idea for something that might work. Ill merge the changes asap.

**Pull request merged**